app/code/Magento/Sales/Controller/Adminhtml/Order/Status/Save.php

```diff
@@ -37,35 +41,55 @@ public function execute()
                 $label = $filterManager->stripTags($label);
             }
 
-            $status = $this->_objectManager->create(\Magento\Sales\Model\Order\Status::class)->load($statusCode);
-            // check if status exist
-            if ($isNew && $status->getStatus()) {
-                $this->messageManager->addError(__('We found another order status with the same order status code.'));
-                $this->_getSession()->setFormData($data);
-                return $resultRedirect->setPath('sales/*/new');
-            }
-
-            $status->setData($data)->setStatus($statusCode);
-
-            try {
-                $status->save();
-                $this->messageManager->addSuccess(__('You saved the order status.'));
-                return $resultRedirect->setPath('sales/*/');
-            } catch (\Magento\Framework\Exception\LocalizedException $e) {
-                $this->messageManager->addError($e->getMessage());
-            } catch (\Exception $e) {
-                $this->messageManager->addException(
-                    $e,
-                    __('We can\'t add the order status right now.')
-                );
+            if ($this->updateStatus($isNew, $data, $statusCode) !== false) {
+                return $this->resultRedirect;
```

Comment:
Such logic seems quite confusing to me.

Random piece of code is moved to a separate method which may return boolean or object :)

Please change method name to `isStatusUpdated` and return `true` instead of object there.